observatory-platform/observatory/platform/utils/template_utils.py

```diff
@@ -274,29 +322,73 @@ def bq_load_shard_v2(project_id: str, transform_bucket: str, transform_blob: str
     :param schema_version: Schema version.
     :param dataset_description: The description for the dataset
+    :param partition_type: The partitioning type (hour, day, month or year)
     :return: None.
     """
     _, bucket_name, data_location, schema_file_path = prepare_bq_load(dataset_id, main_table_id, end_date, prefix,
                                                                       schema_version, dataset_description)
 
     uri = f"gs://{bucket_name}/{transform_blob}"
 
+    # Include date in table id, so data in table is not overwritten
+    partition_table_id = create_date_table_id(partition_table_id, pendulum.today(), partition_type)
     success = load_bigquery_table(uri, dataset_id, data_location, partition_table_id, schema_file_path,
-                                  SourceFormat.NEWLINE_DELIMITED_JSON, partition=True,
-                                  require_partition_filter=False, **load_bigquery_table_kwargs)
+                                  source_format, partition=True, partition_type=partition_type,
+                                  **load_bigquery_table_kwargs)
+    if not success:
+        raise AirflowException()
+
+
+def bq_load_partition(project_id: str, transform_bucket: str, transform_blob: str, dataset_id: str,
+                      dataset_location: str, table_id: str, release_date: pendulum.Pendulum,
+                      source_format: SourceFormat,
+                      partition_type: bigquery.TimePartitioningType, prefix: str = '', schema_version: str = None,
+                      dataset_description: str = '', **load_bigquery_table_kwargs):
+    """ Load data from a specific file (blob) in the transform bucket to a partition.
+
+    :param project_id: project id.
+    :param transform_bucket: transform bucket name.
+    :param transform_blob: Name of the transform blob.
+    :param dataset_id: Dataset id.
+    :param dataset_location: location of dataset.
+    :param table_id: Table id.
+    :param release_date: Release date.
+    :param source_format: the format of the data to load into BigQuery.
+    :param partition_type: The partitioning type (hour, day, month or year)
+    :param prefix: The prefix for the schema.
+    :param schema_version: Schema version.
+    :param dataset_description: description of the BigQuery dataset.
+    :return: None.
+    """
+
+    schema_file_path = prepare_bq_load_v2(project_id, dataset_id, dataset_location, table_id, release_date,
+                                          prefix,
+                                          schema_version, dataset_description)
+
+    uri = f"gs://{transform_bucket}/{transform_blob}"
+
+    # Include date in table id, so data in table is not overwritten
+    table_id = create_date_table_id(table_id, release_date, partition_type)
+    success = load_bigquery_table(uri, dataset_id, dataset_location, table_id, schema_file_path,
+                                  source_format, partition=True, partition_field=partition_field,
```

Comment:
This should come from a parameter in the function, as you may want to change what field is the partition for a specific table.